common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -206,6 +209,24 @@ def handle_ajax(self, dispatch, data):  # TODO: bounds checking
                 self.position = 1
             return json.dumps({'success': True})
 
+        if dispatch == 'get_completion':
+            if not self.is_user_authenticated:
+                return None
+
+            completion_waffle = self.runtime.service(self, 'completion_waffle')
```

Comment:
It would be simpler to have the `CompletionService` class define a `visual_progress_enabled()` function function, rather than having to add in a new service (whose only purpose is to call this one function).